user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala

```diff
@@ -49,41 +52,69 @@ object UserOrigin {
+                      IOResult
+                        .attempt(backup.copyTo(source, overwrite = true))
+                        .foldZIO(
+                          err2 => {
+                            // there, we are in big problem: error in rollback too, likely an FS problem, advice admin
+                            val msg =
+                              s"Error when reverting rudder-users.xlm, you will likely need to have a manual action. Backup file is here and won't be deleted automatically: ${backup.pathAsString}. Error was: ${err2.fullMsg}"
+                            ApplicationLoggerPure.Authz.error(msg) *> Unexpected(msg).fail
+                          },
+                          ok => {
+                            effectUioUnit(backup.delete(swallowIOExceptions = true)) *> ApplicationLoggerPure.Authz
+                              .info(s"User file correctly roll-backed") *> Unexpected(
+                              s"And error happened when trying to save rudder-user.xml file, backup version was restore. Error was: ${err.fullMsg}"
+                            ).fail
+                          }
+                        ),
+                    // in case of update success, we just delete the backup file
+                    ok => effectUioUnit(backup.delete(swallowIOExceptions = true))
+                  )
+      } yield ()
+    }
 
-    if (!file.isWritable) {
-      Failure(s"${file.path} is not writable")
-    } else if (!file.isReadable) {
-      Failure(s"${file.path} is not readable")
-    } else {
-      val p = new RudderPrettyPrinter(300, 4)
-      file.clear()
-      currentXml.foreach { x =>
-        if (x.label == "authentication")
-          file << p.format(newXml)
-        else
-          file << p.format(x)
-      }
-      if (file.contentAsString contains newXml.text)
-        Failure(s"'${file.path}' modification have failed")
-      else
-        Full(file)
+    withBackup(file) {
+      for {
+        writable <- IOResult.attempt(file.isWritable)
+        _        <- ZIO.when(!writable)(Unexpected(s"${file.path} is not writable").fail)
+        readable <- IOResult.attempt(file.isReadable)
+        _        <- ZIO.when(!readable)(Unexpected(s"${file.path} is not readable").fail)
+        p         = new RudderPrettyPrinter(300, 4)
+        _        <- IOResult.attempt(file.clear())
+        _        <- IOResult.attempt {
+                      currentXml.foreach { x =>
+                        if (x.label == "authentication") {
+                          file << p.format(newXml)
+                        } else {
+                          file << p.format(x)
+                        }
+                      }
+                    }
+      } yield ()
     }
   }
 
-  def getUserFilePath: Either[UserConfigFileError, File] = {
-    val resources: Either[UserConfigFileError, UserFile] = UserFileProcessing.getUserResourceFile()
-    resources match {
-      case Left(err) =>
-        Left(err)
```

Comment:
when you see that you write `case Left(err) => Left(err)`, you want to just `map`